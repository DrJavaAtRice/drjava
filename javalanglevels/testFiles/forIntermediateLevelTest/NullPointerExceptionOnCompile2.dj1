//This should also not throw a null pointer exception on compile: it was before, because we weren't handling the ordering of interfaces and classes correctly.

interface TestI2 {
  int silly();
}


class TestYeah2 implements TestI2 {
  int myMethod() { 
    return 5;
  }
  public int silly() {
    return 6; 
  }
}


class TestYou2 {
  int brokenMethod() {
    return 4;//myMethod();
  }
}
