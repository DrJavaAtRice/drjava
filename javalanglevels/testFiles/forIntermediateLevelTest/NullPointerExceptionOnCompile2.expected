// Language Level Converter line number map: dj*->java. Entries: 22
//     1->5         2->6         3->7         4->8         5->9         6->10        7->11        8->12   
//     9->13       10->14       11->15       12->16       13->17       14->18       15->19       16->50   
//    17->51       18->52       19->53       20->54       21->55       22->56   
//This should also not throw a null pointer exception on compile: it was before, because we weren't handling the ordering of interfaces and classes correctly.

interface TestI2 {
  public int silly();
}


class TestYeah2 implements TestI2 {
  public int myMethod() { 
    return 5;
  }
  public int silly() {
    return 6; 
  }

  /** This method is automatically generated by the Language Level Converter. */
  public TestYeah2() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      TestYeah2 cast = ((TestYeah2) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}


class TestYou2 {
  public int brokenMethod() {
    return 4;//myMethod();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public TestYou2() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      TestYou2 cast = ((TestYou2) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}
